.github/workflows/buildcomponents.yaml

```diff
@@ -36,14 +36,12 @@ jobs:
           echo "go_modcache=$(go env GOMODCACHE)" >> $GITHUB_ENV
       - name: Set up cache
         # https://github.com/actions/setup-go/issues/358 - cache is not working with setup-go for multiple jobs
-        uses: actions/cache@v4
+        uses: actions/cache/restore@v4
         with:
           path: |
             ${{ env.go_cache }}
             ${{ env.go_modcache }}
-          key: ${{ env.cache_name }}-${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}-${{ hashFiles('**/go.mod') }}
-          restore-keys: |
-            ${{ env.cache_name }}-${{ runner.os }}-go-
+          key: ${{ env.cache_name }}-${{ runner.os }}-go-
         env:
           cache_name: buildcomponents-go-cache
```

Comment:
```suggestion
          cache_name: run-tests-go-cache # needs to be the same key in the end as in the build step
```

or am I wrong?

Reply:
This is still on the old setup so its a bit of a mess, but since https://github.com/open-component-model/ocm/pull/996 is decoupled from this one, it would still have its own cache. (thats why its buildcomponents, not run-tests)

Reply:
Just read through this again and youre right even within this PR it doesnt make sense. This PR shouldnt have this code for caching in the first place.